tests/validation/tests/v3_api/test_encryption_secrets.py

```diff
@@ -0,0 +1,48 @@
+from .common import *  # NOQA
+import pytest
+from .test_rke_cluster_provisioning import rke_config
+
+providers = [{"aescbc": {"keys": [{"name": "k-fw5hn", "secret": "RTczRjFDODMwQzAyMDVBREU4NDJBMUZFNDhCNzM5N0I="}]}}, {"identity": {}}]
+custom_resources = [{"resources": ["secrets"], "providers": providers}]
+custom_config = {"apiVersion": "apiserver.config.k8s.io/v1", "kind": "EncryptionConfiguration", "resources": custom_resources}
+SECRET_ENCRYPTION_CONFIG = {"customConfig": custom_config, "enabled": True, "type": "/v3/schemas/secretsEncryptionConfig"}
+
+def test_create_custom_cluster_with_encryption():
+    # Creates custom cluster
+    aws_nodes = \
+        AmazonWebServices().create_multiple_nodes(
+            3, random_test_name("jenkins-custom-secret-encryption"))
+    node_roles = [["controlplane"], ["etcd"],
+                  ["worker"]]
+    client = rancher.Client(url=CATTLE_TEST_URL + "/v3",
+                            token=USER_TOKEN, verify=False)
+    cluster = client.create_cluster(
+        name=random_name(),
+        driver="rancherKubernetesEngine",
+        rancherKubernetesEngineConfig=rke_config)
```

Comment:
Can we add the secrets encryption config to the rke config here ?

Reply:
Addressed in changes. Cluster is provisioned with secrets encryption config instead of edited.